### PR TITLE
VM has wrong network statistics with multiple nics in shared networks

### PR DESCRIPTION
**Pull request by ravening**
### Description
If vm has multiple nics belonging to different shared networks then
wrong statistics will be collected since network id is not considred
as primary key. Make the change so that primary key contains network
id so that traffic belonging to that corresponding network is shown

If network id is not added to primary key then all the traffic of all
shared networks will show up in one nic.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [X] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [X] Minor
- [ ] Trivial

### CODE REVIEW

**Review by weizhouapache — APPROVED** [review]
tested ok

**Comment by DaanHoogland** [comment]
@ravening you want this on master only ? not on older branches?

File: usage/src/main/java/com/cloud/usage/parser/NetworkUsageParser.java [hunk 1]
```diff
@@ -80,7 +80,7 @@ public static boolean parse(AccountVO account, Date startDate, Date endDate) {
             long zoneId = usageNetwork.getZoneId();
             String key = "" + zoneId;
             if (usageNetwork.getHostId() != 0) {
-                key += "-Host" + usageNetwork.getHostId();
+                key += "-Host" + usageNetwork.getHostId() + "-Network-" + usageNetwork.getNetworkId();
```
Comment (DaanHoogland) [inline usage/src/main/java/com/cloud/usage/parser/NetworkUsageParser.java:83]: should networkid be checked for 0 as well?

**Comment by ravening** [comment]
> @ravening you want this on master only ? not on older branches?

@DaanHoogland will rebase it to 4.15. will rsolve conflicts as well

**Comment by rhtyd** [comment]
ping @ravening

**Comment by rhtyd** [comment]
ping @ravening please address conflicts and any outstanding issues

File: usage/src/main/java/com/cloud/usage/parser/NetworkUsageParser.java (hunk 1 above)
Comment (ravening) [reply usage/src/main/java/com/cloud/usage/parser/NetworkUsageParser.java:83]: @DaanHoogland this was not needed for networkid 0. the issue was only for extra nics.

**Comment by ravening** [comment]
> ping @ravening please address conflicts and any outstanding issues

@rhtyd fixed it

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Review by DaanHoogland — DISMISSED** [review]
clgtm

**Comment by weizhouapache** [comment]
@blueorangutan test

**Review by RodrigoDLopez — APPROVED** [review]
CLGTM.
Do we need manual tests?

File: engine/schema/src/main/resources/META-INF/db/schema-41500to41510.sql [hunk 2]
```diff
@@ -132,3 +132,7 @@ INSERT INTO `cloud`.`guest_os_hypervisor` (uuid,hypervisor_type, hypervisor_vers
 
 -- Add support for windows2019srvNext_64Guest from VMware 7.0.1.0
 INSERT INTO `cloud`.`guest_os_hypervisor` (uuid,hypervisor_type, hypervisor_version, guest_os_name, guest_os_id, created, is_user_defined) VALUES (UUID(),'VMware', '7.0.1.0', 'windows2019srvNext_64Guest', 276, now(), 0);
+
+ALTER TABLE cloud_usage.usage_network DROP PRIMARY KEY, ADD PRIMARY KEY (`account_id`,`zone_id`,`host_id`,`network_id`,`event_time_millis`);
```
Comment (shwstppr) [inline engine/schema/src/main/resources/META-INF/db/schema-41500to41510.sql:136]: @ravening these DB changes need to moved to 4.16 upgrade path?

**Review by shwstppr — CHANGES_REQUESTED** [review]

File: engine/schema/src/main/resources/META-INF/db/schema-41500to41510.sql (hunk 2 above)
Comment (DaanHoogland) [reply engine/schema/src/main/resources/META-INF/db/schema-41500to41510.sql:136]: good catch @shwstppr.

**Comment by DaanHoogland** [comment]
> CLGTM.
> Do we need manual tests?

@RodrigoDLopez always, but preferably/even better a coded integration test.

**Comment by RodrigoDLopez** [comment]
> @RodrigoDLopez always, but preferably/even better a coded integration test.
I think that it's not easy to do a automated test for those methods. 
I'll run some manual tests here, and put my results as soon as possible.

File: engine/schema/src/main/resources/META-INF/db/schema-41500to41510.sql (hunk 2 above)
Comment (ravening) [reply engine/schema/src/main/resources/META-INF/db/schema-41500to41510.sql]: @shwstppr thanks for pointing it out. have changed it

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@shwstppr are you satisfied with this?

**Review by shwstppr — APPROVED** [review]

**Comment by nvazquez** [comment]
@blueorangutan package

**Review by davidjumani — APPROVED** [review]

**Comment by weizhouapache** [comment]
@ravening can you move the sql change to engine/schema/src/main/resources/META-INF/db/schema-41520to41600.sql ?

**Comment by rhtyd** [comment]
Ping @ravening can you fix conflicts?

**Comment by ravening** [comment]
> Ping @ravening can you fix conflicts?

@rhtyd @weizhouapache done

**Comment by shwstppr** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
@blueorangutan test

**Pull request merged**